synapse/tests/test_lib_stormhttp.py

```diff
@@ -20,3 +20,66 @@ class StormHttpTest(s_test.SynTest):
             opts = {'vars': {'port': port}}
             nodes = await core.nodes(text, opts=opts)
             self.len(1, await core.nodes('test:str=inet:ipv4'))
+
+    async def test_storm_http_post_api(self):
+
+        async with self.getTestCore() as core:
+            core.insecure = True
+            addr, port = await core.addHttpPort(0)
+
+            adduser = '''
+                $url = $lib.str.format("http://127.0.0.1:{port}/api/v1/auth/adduser", port=$port)
+                $user = $lib.dict(name=$name, passwd=$passwd)
+                $post = $lib.inet.http.post($url, json=$user).json().result.name
+                $lib.print($post)
+                [ test:str=$post ]
+            '''
+            opts = {'vars': {'port': port, 'name': 'foo', 'passwd': 'bar'}}
+            nodes = await core.storm(adduser, opts=opts).list()
+            self.len(1, nodes)
+            self.assertIn('foo', [u.name for u in core.auth.users()])
+
+            adduser = '''
+                $url = $lib.str.format("http://127.0.0.1:{port}/api/v1/auth/adduser", port=$port)
+                $user = $lib.str.format('{"name": "{name}", "passwd": "{passwd}"}', name=$name, passwd=$passwd)
+                $header = $lib.dict("Content-Type"="application/json")
```

Comment:
This line becomes un-necessary if we make the tweak to post() above

Reply:
It's actually entirely artificial. The handler being posted too doesn't do content type detection and just treats the body as json.

Reply:
I added this line as more example of how you would pass headers, since I couldn't really find any pure storm examples anywhere.